extensions/skyportal/skyportal/handlers/api/alert.py

```diff
@@ -493,12 +517,209 @@ def post_alert(
+        Include all fields in database. Defaults to False.
+    candid : Int
+        Candidate ID to query
+    """
+
+    # grabbing alerts by single objectId
+    query = {
+        "query_type": "aggregate",
+        "query": {
+            "catalog": "ZTF_alerts",
+            "pipeline": [
+                {
+                    "$match": {
+                        "objectId": object_id,
+                        "candidate.programid": {"$in": program_id_selector},
+                    }
+                },
+                {
+                    "$project": projection
+                    if not include_all_fields
+                    else {
+                        "_id": 0,
+                        "cutoutScience": 0,
+                        "cutoutTemplate": 0,
+                        "cutoutDifference": 0,
+                    }
+                },
+            ],
+        },
+        "kwargs": {"max_time_ms": 10000},
+    }
+
+    if candid is not None:
+        query["query"]["pipeline"][0]["$match"]["candid"] = int(candid)
+
+    response = kowalski.query(query=query)
+
+    if response.get("status", "error") == "success":
+        return response.get("data")
+    else:
+        return None
+
+
+def get_alerts_by_ids(
+    object_ids, program_id_selector, projection=None, include_all_fields=False
+):
+    """Get alert from database (by object IDs).
+    object_ids : List[str]
+        Object IDs to query
+    program_id_selector : List[Int]
+        List of Program IDs to query
+    projection : List[str]
+        List of fields to query from database
+    include_all_fields : bool
+        Include all fields in database. Defaults to False.
+    """
+
+    # otherwise, run a find query with the specified filter
+    query = {
+        "query_type": "find",
+        "query": {
+            "catalog": "ZTF_alerts",
+            "filter": {
+                "objectId": {"$in": [oid.strip() for oid in object_ids.split(",")]},
+                "candidate.programid": {"$in": program_id_selector},
+            },
+            "projection": projection
+            if not include_all_fields
+            else {
+                "_id": 0,
+                "cutoutScience": 0,
+                "cutoutTemplate": 0,
+                "cutoutDifference": 0,
+            },
+        },
+        "kwargs": {
+            "max_time_ms": 10000,
+            "limit": 10000,
+        },
+    }
+
+    response = kowalski.query(query=query)
+
+    if response.get("status", "error") == "success":
+        alert_data = response.get("data")
+        return alert_data
+    else:
+        return None
+
+
+def get_alerts_by_position(
+    ra,
+    dec,
+    radius,
+    radius_units,
+    program_id_selector,
+    projection=None,
+    include_all_fields=False,
+    object_ids=None,
+):
+    """Get alert from database (by object ID).
+    ra : float
+        Right Ascension
+    dec : float
+        Declination
+    radius : float
+        Radius to search
```

Comment:
Specify the units (in the case of radius, say that units can be selected separately but specify what is the default).

Reply:
Good point.